doc/en/fixture.rst

```diff
@@ -322,6 +322,128 @@ The fixtures requested by ``test_foo`` will be instantiated in the following ord
 5. ``f2``: is the last ``function``-scoped fixture in ``test_foo`` parameter list.
 
 
+.. _autoparam:
+
+Combining parametrization and autouse: how to parametrize entire scopes
+-----------------------------------------------------------------------
+
+.. versionadded:: 3.8
+
+Parametrization is a powerful tool for running tests multiple ways with
+different sets of data. But without ``autouse``, they will only be parametrizing
+the fixtures and tests that depend on them; anything the parametrized fixtures
+depended on wouldn't be impacted by the parameter sets, even if they shared the
+same scope. By have a parametrized fixture be autouse, you are telling pytest
+that you want that fixture to impact everything in its designated scope, even
+the fixtures of the same scope that happen before it.
+
+Here's a simple example:
+
+.. code-block:: python
```

Comment:
after my initial skim - if this is the example, then this change is certainly the completely wrong thing to do,
i will defer a deeper review as i expect it to take between 2 and 3 hours to get to the depths of it

but right now i believe merging this as is would be a **major** regression and a impossibly bad solution for parameterization scope

Reply:
I'd love to hear your thoughts on this. Do you think being able to target the parameterized psuedo-scope directly (and maybe having it bubble up to the full scope if no parameterization is used, so it can still apply to both of the classes in this example) would be a better option? Or maybe having the fixture be able to define some other option so it can still control the parameterization of the whole scope and other fixtures don't have to worry about it? Or prerhaps something else entirely?

I think I could modify what I've done so far fairly easily to accommodate for something like that.

Reply:
@SalmonMode its not clear to me yet - based on what i see its a major regression, and its not clear how to fix it

i would also like to note, that the current underpinnings of fixtures, their caching and scopes are so badly defined and structured, that any regression this one brings in will likely be unfixable without removing the feature

the last time we tried something like it was for multi-scoped fixtures just before pytest 3.0, and it failed harshly due to the unexpected behaviors in the underpinnings

Reply:
Sorry, just a little confused. Are you saying that your issue is only with this implementation and the risks it brings? Or is the core idea of this change (i.e. parameterization + autouse = parameterize the whole scope) still an issue for you?

Reply:
@SalmonMode the implementation as is breaks the api and is a major regression, then there is a risk

the core idea of this change will basically break existing testsuites and all expectations for the very basics of how scopes work

Reply:
we don't actually have tests that capture the dependency handling of fixtures in such detail as to express what this change would break - in addition its creating a situation where the caching and time cost behavior suddenly has actions at a distance - i strongly object those actions at a distance - its just wrong